lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -681,6 +678,42 @@ protected function prepareUpdateData($entity)
         return $result;
     }
 
+    /**
+     * Decides if an extra update is required for the entity being persisted
+     * This is only required if the associated entity is different from the current one,
+     * and if the current entity relies on the database to generate its id
+     *
+     * @param mixed $entity
+     * @param mixed $newVal
+     *
+     * @return bool Returns true is an extra update is required
+     */
+    private function isExtraUpdateRequired($entity, $newVal) : bool
+    {
+        if ($newVal === null) {
+            return false;
+        }
+        $oid = spl_object_hash($newVal);
+        $uow = $this->em->getUnitOfWork();
+        if (isset($this->queuedInserts[$oid]) || $uow->isScheduledForInsert($newVal)) {
```

Comment:
You could negate this condition and use an early return.

Reply:
fixed